kolibri/core/assets/src/views/breadcrumbs/index.vue

```diff
@@ -0,0 +1,254 @@
+        >
+          <span :style="{ maxWidth: `${lastCrumbMaxWidth}px` }">{{ crumb.text }}</span>
+        </li>
+      </template>
+    </ol>
+
+  </nav>
+
+</template>
+
+
+<script>
+
+  const ResizeSensor = require('css-element-queries/src/ResizeSensor');
+  const ResponsiveElement = require('kolibri.coreVue.mixins.responsiveElement');
+  const ValidateLinkObject = require('kolibri.utils.validateLinkObject');
+  const filter = require('lodash/filter');
+  const startsWith = require('lodash/startsWith');
+  const throttle = require('lodash/throttle');
+
+  const DROPDOWN_BTN_WIDTH = 55;
+  const DROPDOWN_SIDE_PADDING = 32; // pulled from .breadcrumbs-dropdown
+  const MAX_CRUMB_WIDTH = 300; // pulled from .breadcrumbs-visible-item class
+
+  module.exports = {
+    $trNameSpace: 'breadcrumbs',
+
+    mixins: [ResponsiveElement],
+
+    components: {
+      'ui-icon-button': require('keen-ui/src/UiIconButton'),
+    },
+
+    props: {
+      items: {
+        type: Array,
+        required: true,
+        validator(crumbItems) {
+          const crumbs = Array.from(crumbItems);
+          // Must not be empty
+          if (!crumbs.length) {
+            return false;
+          }
+          // All must have text
+          if (!crumbs.every(crumb => Boolean(crumb.text))) {
+            return false;
+          }
+          crumbs.pop();
+          // All, but the last, must have a valid router link
+          return crumbs.every(crumb => ValidateLinkObject(crumb.link));
+        },
+      },
+    },
+
+    data: () => ({
+      crumbs: [],
+    }),
+
+    computed: {
+      collapsedCrumbs() {
+        return this.crumbs.filter(crumb => crumb.collapsed === true).reverse();
+      },
+
+      parentWidth() {
+        return this.elSize.width;
+      },
+
+      lastCrumbMaxWidth() {
+        if (this.collapsedCrumbs.length) {
+          return Math.min(this.parentWidth - DROPDOWN_BTN_WIDTH, MAX_CRUMB_WIDTH);
+        }
+        return Math.min(this.parentWidth, MAX_CRUMB_WIDTH);
+      },
+      collapsedCrumbMaxWidth() {
+        return Math.min(this.parentWidth - DROPDOWN_SIDE_PADDING, MAX_CRUMB_WIDTH);
+      },
+    },
+
+    methods: {
+      attachSensors() {
+        this.$nextTick(() => {
+          const crumbRefs = filter(this.$refs, (value, key) => startsWith(key, 'crumb'));
+          this.crumbs = this.crumbs.map((crumb, index) => {
+            const updatedCrumb = crumb;
+            updatedCrumb.ref = crumbRefs[index];
+            updatedCrumb.sensor = new ResizeSensor(updatedCrumb.ref, () => {
+              this.throttleUpdateCrumbs();
+            });
+            return updatedCrumb;
+          });
+          this.updateCrumbs();
+        });
+      },
+
+      resetCollapsedState() {
+        this.crumbs = this.crumbs.map(crumb => {
+          const updatedCrumb = crumb;
+          updatedCrumb.collapsed = false;
+          return updatedCrumb;
+        });
+      },
+
+      updateCrumbs() {
```

Comment:
We discussed this a bit in person:

For a future enhancement, I believe it might be possible to get rid of the `resetCollapsedState`, `$nextTick`, and 250ms throttle by rendering all crumbs at full width offscreen. Those offscreen items become the reference widths, and then the rest of this `updateCrumbs` function should become cheap enough to run much more often. It might also permit some more sophisticated logic, like relaxing the max width constraint under certain conditions.